src/Components/Modal/Modal.jsx

```diff
@@ -0,0 +1,104 @@
+import "./modal.css";
+import { useState } from "react";
+import { v4 as uuid } from "uuid";
+import { useNotes } from "../../Context/noteContext";
+import { getCurrentDate } from "../../helper";
+import { tagData } from "../../helper";
+
+const Modal = ({ modalClose }) => {
+  const { noteDispatch } = useNotes();
+  const { setShowModal } = modalClose;
+  const [title, setTitle] = useState("");
+  const [content, setContent] = useState("");
+  const [taglist, setTagList] = useState([]);
+  const handleTagList = (e) => {
+    if (e.target.checked) {
+      taglist.includes(e.target.value)
+        ? setTagList([...taglist])
+        : setTagList([...taglist, e.target.value]);
+    } else {
+      setTagList([...taglist.filter((item) => item !== e.target.value)]);
+    }
+  };
+
+  const saveData = () => {
+    const newnote = {
+      _id: uuid(),
+      noteTitle: title === "" ? "Empty Note" : title,
+      noteContent: content,
+      noteTags: taglist,
+      date: getCurrentDate(),
+    };
+    setShowModal((showModal) => !showModal);
+    setTitle("");
+    setContent("");
+    setTagList([]);
+    noteDispatch({ type: "ADD_TO_ALL_NOTES", payload: newnote });
+  };
+  return (
+    <div
+      className="input-modal-body"
+      onClick={() => setShowModal((showModal) => !showModal)}
+    >
+      <div className="modal input-modal" onClick={(e) => e.stopPropagation()}>
```

Comment:
e.stopPropagation is used here so that the state change of modal which is applied on the parent div (which makes the modal close) is not carried over to the modal content i.e. the modal should not close on clicking on it.